frontend/src/container/GridGraphLayout/Graph/FullView/GraphManager.tsx

```diff
@@ -0,0 +1,326 @@
+			if (index !== -1) {
+				graphDisplayStatusArray[i] = legendEntries[index].show;
+			}
+		});
+		setGraphVisibilityArray(graphDisplayStatusArray);
+	}, [data, legendEntries]);
+
+	useEffect(() => {
+		if (localStorage.getItem('LEGEND_GRAPH') !== null) {
+			const legendGraphFromLocalStore = localStorage.getItem('LEGEND_GRAPH');
+			const legendFromLocalStore: [
+				{ name: string; dataIndex: LegendEntryProps[] },
+			] = JSON.parse(legendGraphFromLocalStore as string);
+			let isfound = false;
+			const graphDisplayStatusArray = Array(data.datasets.length).fill(true);
+			legendFromLocalStore.forEach((item) => {
+				if (item.name === name) {
+					setLegendEntries(item.dataIndex);
+					data.datasets.forEach((d, i) => {
+						const index = item.dataIndex.findIndex((di) => di.label === d.label);
+						if (index !== -1) {
+							graphDisplayStatusArray[i] = item.dataIndex[index].show;
+						}
+					});
+					setGraphVisibilityArray(graphDisplayStatusArray);
+					isfound = true;
+				}
+			});
+			if (!isfound) {
+				setGraphVisibilityArray(Array(data.datasets.length).fill(true));
+				setLegendEntries(showAllDataSet(data));
+			}
+		} else {
+			setGraphVisibilityArray(Array(data.datasets.length).fill(true));
+			setLegendEntries(showAllDataSet(data));
+		}
+	}, [data, name]);
+
+	const checkBoxOnChangeHandler = (
+		e: CheckboxChangeEvent,
+		index: number,
+	): void => {
+		graphVisibilityArray[index] = e.target.checked;
+		setGraphVisibilityArray([...graphVisibilityArray]);
+		if (graphVisibilityHandler) {
+			graphVisibilityHandler(graphVisibilityArray);
+		}
+	};
+
+	const getCheckBox = (index: number): React.ReactElement => (
+		<ConfigProvider
+			theme={{
+				token: {
+					colorPrimary: data.datasets[index].borderColor?.toString(),
+					colorBorder: data.datasets[index].borderColor?.toString(),
+					colorBgContainer: data.datasets[index].borderColor?.toString(),
+				},
+			}}
+		>
+			<Checkbox
+				onChange={(e): void => checkBoxOnChangeHandler(e, index)}
+				checked={graphVisibilityArray[index]}
+			/>
+		</ConfigProvider>
+	);
+
+	const labelClickedHandler = (labelIndex: number): void => {
+		const newGraphVisibilityArray = Array(data.datasets.length).fill(false);
+		newGraphVisibilityArray[labelIndex] = true;
+		setGraphVisibilityArray([...newGraphVisibilityArray]);
+		if (graphVisibilityHandler) {
+			graphVisibilityHandler(newGraphVisibilityArray);
+		}
+	};
+
+	const getLabel = (label: string, labelIndex: number): React.ReactElement => {
+		let newLebal = label;
+		if (label.length > 30) {
+			newLebal = `${label.substring(0, 30)}...`;
+		}
+		return (
+			<button
+				type="button"
+				style={{
+					maxWidth: '300px',
+					cursor: 'pointer',
+					border: 'none',
+					backgroundColor: 'transparent',
+					color: 'white',
+				}}
+				onClick={(): void => labelClickedHandler(labelIndex)}
+			>
+				{newLebal}
+			</button>
+		);
+	};
+
+	const columns: ColumnType<DataSetProps>[] = [
```

Comment:
why are not using the util to form the column? there should be a base config for this one